src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -0,0 +1,491 @@
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            Int32 offsetMinutes = (Int32)value.Offset.TotalMinutes;
+
+            switch (dateTimeFormat)
+            {
+                case ODataJsonDateTimeFormat.ISO8601DateTime:
+                    {
+                        // Uses the same format as DateTime but with offset:
+                        // jsonDateTime= quotation-mark
+                        //  YYYY-MM-DDThh:mm:ss.sTZD
+                        //  [("+" / "-") offset]
+                        //  quotation-mark
+                        //
+                        // offset = 4DIGIT
+                        string textValue = XmlConvert.ToString(value);
+                        await writer.WriteQuotedAsync(textValue).ConfigureAwait(false);
+                    }
+
+                    break;
+
+                case ODataJsonDateTimeFormat.ODataDateTime:
+                    {
+                        // Uses the same format as DateTime but with offset:
+                        // jsonDateTime= quotation-mark
+                        //  "\/Date("
+                        //  ticks
+                        //  [("+" / "-") offset]
+                        //  ")\/"
+                        //  quotation-mark
+                        //
+                        // ticks = *DIGIT
+                        // offset = 4DIGIT
+                        string textValue = String.Format(
+                            CultureInfo.InvariantCulture,
+                            JsonConstants.ODataDateTimeOffsetFormat,
+                            DateTimeTicksToJsonTicks(value.Ticks),
+                            offsetMinutes >= 0 ? JsonConstants.ODataDateTimeOffsetPlusSign : string.Empty,
+                            offsetMinutes);
+                        await writer.WriteQuotedAsync(textValue).ConfigureAwait(false);
+                    }
+
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Write a TimeSpan value.
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">TimeSpan value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, TimeSpan value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteQuotedAsync(EdmValueWriter.DurationAsXml(value)).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Write a TimeOfDay value
+        /// </summary>
+        /// <param name="writer">The text writer to write the output to.</param>
+        /// <param name="value">TimeOfDay value to be written.</param>
+        internal static async Task WriteValueAsync(this TextWriter writer, TimeOfDay value)
+        {
+            Debug.Assert(writer != null, "writer != null");
+
+            await writer.WriteQuotedAsync(value.ToString()).ConfigureAwait(false);

```

Comment:
Add CultureInfo

Reply:
Here we're formatting a `TimeOfDay` value and that is taken care of in the `Edm` library [here](https://github.com/OData/odata.net/blob/cc9d86f0069be395a4f07aabe5cad0cec7b9a2cd/src/Microsoft.OData.Edm/Schema/TimeOfDay.cs#L297). The `.ToString()` method for `TimeOfDay` does not support a `CultureInfo` argument

Reply:
Actually this line is taking care of the CultureInfo https://github.com/OData/odata.net/blob/cc9d86f0069be395a4f07aabe5cad0cec7b9a2cd/src/Microsoft.OData.Edm/Schema/TimeOfDay.cs#L302